src/log.js

```diff
@@ -747,23 +747,55 @@ class MemLogger extends FormattedLoggerBase {
  *
  * @class
  * @param {Object} opts – Optional, named parameters
- * @param {string} [opts.level='silly'] The minimum log level to sent to loggly
+ * @param {string} [opts.level='silly'] The minimum log level to sent to the logger
+ * @param {Logger} [opts.logger = rootLogger] The helix logger to use
  * @param {Function} [opts.filter=identity] Will be given every log message to perform
  *   arbitrary transformations; must return either another valid message object or undefined
  *   (in which case the message will be dropped).
+ * @param {object} [opts.fields] Additional log fields to add to every log message.
  */
 class InterfaceBase {
+  /**
+   * Helper method for setting defaults recursively.
+   * todo: add to ferrum?
+   * @private
+   */
+  static _deepsetdefault(src, dst) {
+    Object.entries(src).forEach(([key, value]) => {
+      if (key in dst) {
+        if (typeof value === 'object') {
+          InterfaceBase._deepsetdefault(value, dst[key]);
+        }
+      } else {
+        dst[key] = value;
+      }
+    });
+  }
+
   constructor({
-    logger = rootLogger, level = 'silly', filter = identity, ...unknown
+    logger = rootLogger, level = 'silly', filter = identity, fields = {}, ...unknown
   } = {}) {
-    assign(this, { logger, level, filter });
+    assign(this, {
+      logger, level, filter, fields,
+    });
     if (!empty(unknown)) {
       throw new Error(`Unknown named options given to ${typename(type(this))}: ${tryInspect(unknown)}`);
     }
   }
 
+  /**
+   * Helper method that creates a child logging interface that is derived from this logger, merging
+   * the given options.
+   * @param {object} opts Options to merge with this logger
+   */
+  child(opts) {
```

Comment:
Let's please make this a function; we should also give it a more specific name (I still quite like deriveSomethingSomething); `child` is fairly nonspecific and invokes concepts from OOP)

Reply:
ok

Reply:
Still needs to be a function…